File: app/src/main/java/com/legue/axel/lolsummonertool/database/model/summoner/Summoner.java
package com.legue.axel.lolsummonertool.database.model.summoner;

import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;
import androidx.annotation.NonNull;


@Entity(tableName = "summoners")
public class Summoner {

    @PrimaryKey()
    @NonNull
    public String id;
    public String name;
    public String puuid;
    public int summonerLevel;
    public String accountId;
    public long revisionDate;
    public int profileIconId;

    @Ignore
    public Summoner() {
    }

    public Summoner( String id, String name, String puuid, int summonerLevel, String accountId, long revisionDate, int profileIconId) {
        this.id = id;
        this.name = name;
        this.puuid = puuid;
        this.summonerLevel = summonerLevel;
        this.accountId = accountId;
        this.revisionDate = revisionDate;
        this.profileIconId = profileIconId;
    }
}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/model/summoner/Summoner.kt
package com.legue.axel.lolsummonertool.database.model.summoner

import androidx.room.Entity
import androidx.room.PrimaryKey


@Entity(tableName = "summoners")
data class Summoner(
        @PrimaryKey
        var id: String,
        var name: String? = null,
        var puuid: String? = null,
        var summonerLevel: Int = 0,
        var accountId: String,
        var revisionDate: Long = 0,
        var profileIconId: Int = 0
)